abc128/C.rb

```diff
@@ -0,0 +1,32 @@
+# https://atcoder.jp/contests/abc128/tasks/abc128_c
+
+MOD = 2
+
+N, M = gets.split.map(&:to_i)
+bulbs_with_switches = []
+M.times do
+  bulbs_with_switches << gets.split.map(&:to_i)
+end
+ps = gets.split.map(&:to_i)
+
+ans = 0
+(0..(2 ** N - 1)).each do |i|
```

Comment:
本当はこうなんじゃないかな？と思う
```suggestion
(1..(2 ** N - 1)).each do |i|
```
全部スイッチがOFFの時は電球が点灯する訳ないので、対象外にすべきだと思う。

が、対象外にするとWAになる... https://atcoder.jp/contests/abc128/submissions/9919361

Reply:
https://www.dropbox.com/sh/arnpe0ef5wds8cv/AADANyGKdFPfhbTGVx_CA5e8a/ABC128/C

WAのテストケース

---
#### 9.txt
in:
```
10 10
5 7 9 5 10 8
6 8 4 1 6 2 5
6 4 6 2 8 1 5
6 2 8 5 4 6 1
5 9 7 10 8 5
7 10 9 2 4 1 7 6
5 7 10 5 8 9
7 1 6 4 2 9 7 10
5 7 9 5 8 10
7 9 1 7 10 4 6 2
0 0 0 0 0 0 0 0 0 0
```
out:
```
256
```

---
#### 17.txt
in:
```
10 1
3 7 3 6
0
```
out:
```
512
```

---
#### 23.txt
in:
```
10 1
10 1 2 3 4 5 6 7 8 9 10
0
```
out:
```
512
```

Reply:
input の最後の行( p1 ... pM ) がすべてゼロのテストケースだと、スイッチが全OFFの場合に「on になっているスイッチの個数」がすべてゼロになるので、それを「 2 で割った余り」もゼロで、条件を満たす組み合わせとしてカウントされてしまう。